crates/oxc_linter/src/rules/nextjs/no_unwanted_polyfillio.rs

```diff
@@ -0,0 +1,275 @@
+    "Array.prototype.find",
+    "Array.prototype.findIndex",
+    "Array.prototype.flatMap",
+    "Array.prototype.flat",
+    "Array.from",
+    "Array.prototype.includes",
+    "Array.of",
+    "Function.prototype.name",
+    "fetch",
+    "Map",
+    "Number.EPSILON",
+    "Number.Epsilon",
+    "Number.isFinite",
+    "Number.isNaN",
+    "Number.isInteger",
+    "Number.isSafeInteger",
+    "Number.MAX_SAFE_INTEGER",
+    "Number.MIN_SAFE_INTEGER",
+    "Number.parseFloat",
+    "Number.parseInt",
+    "Object.assign",
+    "Object.entries",
+    "Object.fromEntries",
+    "Object.getOwnPropertyDescriptor",
+    "Object.getOwnPropertyDescriptors",
+    "Object.is",
+    "Object.keys",
+    "Object.values",
+    "Reflect",
+    "Set",
+    "Symbol",
+    "Symbol.asyncIterator",
+    "String.prototype.codePointAt",
+    "String.prototype.endsWith",
+    "String.fromCodePoint",
+    "String.prototype.includes",
+    "String.prototype.@@iterator",
+    "String.prototype.padEnd",
+    "String.prototype.padStart",
+    "String.prototype.repeat",
+    "String.raw",
+    "String.prototype.startsWith",
+    "String.prototype.trimEnd",
+    "String.prototype.trimStart",
+    "URL",
+    "URL.prototype.toJSON",
+    "URLSearchParams",
+    "WeakMap",
+    "WeakSet",
+    "Promise",
+    "Promise.prototype.finally",
+    "es2015", // Should be covered by babel-preset-env instead.
+    "es2016", // contains polyfilled 'Array.prototype.includes', 'String.prototype.padEnd' and 'String.prototype.padStart'
+    "es2017", // contains polyfilled 'Object.entries', 'Object.getOwnPropertyDescriptors', 'Object.values', 'String.prototype.padEnd' and 'String.prototype.padStart'
+    "es2018", // contains polyfilled 'Promise.prototype.finally' and ''Symbol.asyncIterator'
+    "es2019", // Contains polyfilled 'Object.fromEntries' and polyfilled 'Array.prototype.flat', 'Array.prototype.flatMap', 'String.prototype.trimEnd' and 'String.prototype.trimStart'
+    "es5", // Should be covered by babel-preset-env instead.
+    "es6", // Should be covered by babel-preset-env instead.
+    "es7", // contains polyfilled 'Array.prototype.includes', 'String.prototype.padEnd' and 'String.prototype.padStart'
+};
+
+impl Rule for NoUnwantedPolyfillio {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::JSXOpeningElement(jsx_el) = node.kind() {
+            let JSXElementName::Identifier(JSXIdentifier { name: tag_name, .. }) = &jsx_el.name
+            else {
+                return;
+            };
+
+            let next_script_import_local_name =
+                ctx.semantic().module_record().import_entries.iter().find_map(|entry| {
+                    if entry.module_request.name().as_str() == "next/script" {
+                        Some(entry.local_name.name())
+                    } else {
+                        None
+                    }
+                });
+
+            if tag_name.as_str() != "script"
```

Comment:
I think this `tag_name.as_str() != "script"` should come before `next_script_import_local_name` for better performance.

We should always do cheap checks and return early before all the heavy lifting.

Reply:
I moved `next_script_import_local_name` to the inside of `tag_name.as_str() != "script"` if statement!